game.cpp

```diff
@@ -1686,14 +1686,15 @@ void test_game() //!OCLINT tests may be many
   }
 #endif
 
-//#define FIX_ISSUE_458
+#define FIX_ISSUE_458
 #ifdef FIX_ISSUE_458
   {
     // (458) The color of any food item can be accessed easily
-    const game g;
-    const color color_food = get_nth_food_color(g, 0);
-    const color default_color;
-    assert(color_food == default_color);
+    game g;
```

Comment:
See for example [https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rp-mutable](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rp-mutable)

Reply:
I agree with @richelbilderbeek. In general, you should not modify tests as part as solving an issue, unless:

1. The test contains a mistake, such as a typo or a misunderstanding of how the code works. This can happen, because mediors are humans and humans have the feature of making mistakes 🤖
2. The test is outdated, because it has been written some time ago, the code has evolved and taken a different direction since, and so the test is unsolvable or does no longer assert what it should assert.

AFAICS, this is not the case here (but I may be missing something?)

Reply:
My guess is that you removed `const` because the compiler didn't like the output of `get_nth_food()` not being const. If so, I believe there is one or more easy ways to ensure that the function returns a const variable. Have a try 🍀

Reply:
I have added const back to the code